rpc/handlers.go

```diff
@@ -627,6 +661,12 @@ func (h *Handler) TransactionReceiptByHash(hash felt.Felt) (*TransactionReceipt,
 		return nil, ErrTxnHashNotFound
 	}
 
+	return h.transactionReceipt(receipt, txn, blockHash, blockNumber)
+}
+
+func (h *Handler) transactionReceipt(receipt *core.TransactionReceipt, txn core.Transaction,
+	blockHash *felt.Felt, blockNumber uint64,
+) (*TransactionReceipt, *jsonrpc.Error) {
```

Comment:
You can pass in the block status here as a parameter as well, so that when this is called in a loop it doesn't end up making multiple `h.l1head` calls unnecessarily to figure out what the txn status is.